Research/Research/RSDDetailInputFieldObject.swift

```diff
@@ -0,0 +1,208 @@
+        case transitionStyle
+        case inputPrompt = "prompt"
+        case inputPromptDetail = "promptDetail"
+        case placeholder
+        case inputUIHint = "uiHint"
+    }
+    
+    /// The transition style for showing the detail editing view.
+    open var transitionStyle: RSDTransitionStyle?
+    
+    /// The prompt to display for the detail. This is the element displayed in the parent view controller
+    /// as button text or above the cell if using a placeholder/formatter style of UI/UX.
+    open var inputPrompt: String?
+    
+    /// The prompt detail is additional text displayed in the parent view controller.
+    open var inputPromptDetail: String?
+    
+    /// The placeholder text to display for a `nil` input field (before the detail has been added).
+    open var placeholder: String?
+    
+    /// The UI hint for displaying the cell (button, link, discloserArrow, etc.).
+    open var inputUIHint: RSDFormUIHint?
+    
+    /// Override to allow decoding of subclass variables.
+    open override func decode(from decoder: Decoder, for deviceType: RSDDeviceType?) throws {
+        try super.decode(from: decoder, for: deviceType)
+        
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.transitionStyle = try container.decodeIfPresent(RSDTransitionStyle.self, forKey: .transitionStyle) ?? self.transitionStyle
+        self.inputPrompt = try container.decodeIfPresent(String.self, forKey: .inputPrompt) ?? self.inputPrompt
+        self.inputPromptDetail = try container.decodeIfPresent(String.self, forKey: .inputPromptDetail) ?? self.inputPromptDetail
+        self.placeholder = try container.decodeIfPresent(String.self, forKey: .placeholder) ?? self.placeholder
+        self.inputUIHint = try container.decodeIfPresent(RSDFormUIHint.self, forKey: .inputUIHint) ?? self.inputUIHint
+    }
+    
+    
+    // MARK: Wrapper for `RSDInputField`
+    
+    /// Whether or not the detail is optional is determined by whether or not all the input fields are
+    /// optional.
+    open var isOptional: Bool {
+        return self.inputFields.reduce(true, { $0 && $1.isOptional })
+    }
+    
+    /// The data type returned is the first input field if there is only one input field include in this step,
+    /// otherwise it is a custom type.
+    open var dataType: RSDFormDataType {
+        if self.inputFields.count == 1 {
+            return self.inputFields.first!.dataType
+        }
+        else {
+            // Use `.string` as the base type b/c anything can be described as a String.
+            return .custom("detail", .string)
+        }
+    }
+    
+    /// Returns `nil` by default.
+    open var textFieldOptions: RSDTextFieldOptions? {
+        return nil
+    }
+    
+    /// Returns `nil` by default.
+    open var range: RSDRange? {
+        return nil
+    }
+    
+    /// Returns the formatter from the first input field (if only one), otherwise will return `nil`.
```

Comment:
semicolon or new sentence 
 -or-
replace the final clause with just "or `nil`."